.github/workflows/auto_release.yaml

```diff
@@ -0,0 +1,71 @@
+name: auto_release
+
+on:
+  push:
+    tags:
+      - 'v*'
+
+jobs:
+  release:
+    name: Publish GitHub release
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Fetch all history for all tags and branches
+        run: git fetch --prune --unshallow
+      - name: Install gcc-mingw-w64-x86-64
+        run: |
+             sudo apt -q update
+             sudo apt -q install --no-install-recommends gcc-mingw-w64-x86-64
+      - name: Build truckersmp-cli.exe
+        run: make
+      - name: Strip truckersmp-cli.exe
+        run: x86_64-w64-mingw32-strip truckersmp-cli.exe
+      - name: Compress files
+        id: compress_files
+        run: |
+             tag="$(basename ${{ github.ref }})"
+             tardir="$(basename ${{ github.repository }})-${tag:1}"
+             mkdir -p .output/$tardir
+             mv * .output/$tardir
+             tar Jcvf output.tar.xz -C .output --owner=root --group=root $tardir
```

Comment:
Arch switched from xz to zst not long ago because of multi threading, reproducibility and more performance. Is that something we should also care about (it's rather small anyway) and how widespread is it on other distributions since you want to look at other OSs in the future?

Reply:
I guess we can *add* an asset archive in .tzst (tar + zstd) format but we cannot *replace* .tar.xz with .tzst.

Self-update (downloading, version comparing, *unpacking archive*, and executing main script, etc.) would be done with Python. However, unfortunately, [as of Python 3.9, zstd is not supported yet](https://docs.python.org/3.9/library/archiving.html) and I would not like to require external Python packages to support zstd format.